Enhancement/5451 twg widget

eugene-manuilov opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #5451

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

techanvil reviewed (changes requested):
Hi @eugene-manuilov, code-wise this is looking good with just a couple of minor comments from me.

However when it comes to testing it, I'm unable to follow the QAB as specified. The reason being, although the QAB mentions simply _activating_ the TwG module, in fact the module needs to be connected in order for the widget to be registered:

https://github.com/google/site-kit-wp/blob/def2970f662ed39fbfdd6c53df87ec11876703f6/includes/Modules/Thank_With_Google.php#L56-L58

At the moment it's not possibly to get as far as connecting the TwG module, unless I am missing something?

So at the moment when I add a Legacy Widget it doesn't give me the option for the TwG: Supporter Wall widget. Whereas if I comment out that check for `is_connected()`, it does show up. Not sure if I have missed something here, or if the QAB needs a rethink, or possibly this should be tested once TwG can be fully connected...

eugene-manuilov commented on includes/Modules/Thank_With_Google/Supporter_Wall_Widget.php:
```diff
@@ -0,0 +1,106 @@
+		$title_id = $this->get_field_id( 'title' );
+
+		echo '<p>';
+			echo '<label for="', esc_attr( $title_id ), '">';
```
Yes, no problem, updated to use the `?> ... <?php` approach. 👍

eugene-manuilov commented on includes/Modules/Thank_With_Google/Supporter_Wall_Widget.php:
```diff
@@ -0,0 +1,106 @@
+	public function __construct() {
+		parent::__construct(
+			self::WIDGET_ID,
+			sprintf( 'Thank with Google: %s', __( 'Supporter Wall', 'google-site-kit' ) )
```
I am not sure about this... As far as I know product names are not translated since they are "international" names. For example, take a look at https://www.thinkwithgoogle.com/ - you can switch between different languages and see that "Think with Google" renames untranslated.

techanvil replied:
From what I can see we are translating the service names, for example -

https://github.com/google/site-kit-wp/blob/cc86db8f12665a5badf9a5bbbe48d20cce087d8b/includes/Modules/Thank_With_Google.php#L117

This can be also seen for other modules and in JS too...

eugene-manuilov replied:
Fair point, I'll update 👍

eugene-manuilov replied:
Updated, should be good now.

eugene-manuilov commented:
> However when it comes to testing it, I'm unable to follow the QAB as specified. The reason being, although the QAB mentions simply _activating_ the TwG module, in fact the module needs to be connected in order for the widget to be registered:
> 
> At the moment it's not possibly to get as far as connecting the TwG module, unless I am missing something?
> 
> So at the moment when I add a Legacy Widget it doesn't give me the option for the TwG: Supporter Wall widget. Whereas if I comment out that check for `is_connected()`, it does show up. Not sure if I have missed something here, or if the QAB needs a rethink, or possibly this should be tested once TwG can be fully connected...

I have added one step to QAB to use a snippet from another ticket to fake TwG setup. Does it help?

techanvil commented:
> > However when it comes to testing it, I'm unable to follow the QAB as specified. The reason being, although the QAB mentions simply _activating_ the TwG module, in fact the module needs to be connected in order for the widget to be registered:
> > At the moment it's not possibly to get as far as connecting the TwG module, unless I am missing something?
> > So at the moment when I add a Legacy Widget it doesn't give me the option for the TwG: Supporter Wall widget. Whereas if I comment out that check for `is_connected()`, it does show up. Not sure if I have missed something here, or if the QAB needs a rethink, or possibly this should be tested once TwG can be fully connected...
> 
> I have added one step to QAB to use a snippet from another ticket to fake TwG setup. Does it help?

Thanks @eugene-manuilov, that did help and I've been able to connect TwG. One thing to note, though is that as well as inserting the PHP snippet I also had to run the JS snippet from the issue's QA Brief, so it might be worth a mention of this. E.g.

```diff
-Add a snippet from the https://github.com/google/site-kit-wp/issues/5450 to fake TwG setup into the functions.php file or into a mu-plugin.
+Follow the Setup instructions in the QA Brief for https://github.com/google/site-kit-wp/issues/5450 to connect the TwG module.
```

eugene-manuilov commented:
> > > However when it comes to testing it, I'm unable to follow the QAB as specified. The reason being, although the QAB mentions simply _activating_ the TwG module, in fact the module needs to be connected in order for the widget to be registered:
> > > At the moment it's not possibly to get as far as connecting the TwG module, unless I am missing something?
> > > So at the moment when I add a Legacy Widget it doesn't give me the option for the TwG: Supporter Wall widget. Whereas if I comment out that check for `is_connected()`, it does show up. Not sure if I have missed something here, or if the QAB needs a rethink, or possibly this should be tested once TwG can be fully connected...
> > 
> > 
> > I have added one step to QAB to use a snippet from another ticket to fake TwG setup. Does it help?
> 
> Thanks @eugene-manuilov, that did help and I've been able to connect TwG. One thing to note, though is that as well as inserting the PHP snippet I also had to run the JS snippet from the issue's QA Brief, so it might be worth a mention of this. E.g.
> 
> ```diff
> -Add a snippet from the https://github.com/google/site-kit-wp/issues/5450 to fake TwG setup into the functions.php file or into a mu-plugin.
> +Follow the Setup instructions in the QA Brief for https://github.com/google/site-kit-wp/issues/5450 to connect the TwG module.
> ```

Yep, sounds good to me. Updated. Thanks.

techanvil reviewed (approved):
LGTM! Nice one @eugene-manuilov
